packages/ws-server/src/webSocketServer.ts

```diff
@@ -42,27 +44,57 @@ const logger = mainLogger.child({ name: 'rpc-server' });
 const register = new Registry();
 const relay: Relay = new RelayImpl(logger, register);
 let connectedClients = 0;
+let clientIps = {};
 const MAX_CONNECTIONS = parseInt(process.env.CONNECTION_LIMIT || '10');
 
-const app = websockify(new Koa(), { 
-        verifyClient: function(info, done) {
-            if (connectedClients >= MAX_CONNECTIONS) {
-                return done(false, 429, 'Connection limit exceeded');
-            }
-            done(true);
+const app = websockify(new Koa(), {
+    verifyClient: function(info, done) {
+        if (connectedClients >= MAX_CONNECTIONS) {
+            return done(false, 429, 'Connection limit exceeded');
         }
+        done(true);
+    }
 });
 
 const LOGGER_PREFIX = 'WebSocket:';
 
 const CHAIN_ID = relay.eth().chainId();
+const IP_LIMIT_ERROR = WebSocketError.CONNECTION_IP_LIMIT_EXCEEDED;
```

Comment:
I agree that this approach is cleaner for handling the connection limits.  I would like to wait until this is merged into main so I can pick it up in this [PR](https://github.com/hashgraph/hedera-json-rpc-relay/pull/1017) and use it.